flake.nix

```diff
@@ -22,37 +32,79 @@
             openssl
             pkg-config
             perl
+            fenix-channel.rustc
           ];
 
           nativeBuildInputs = with pkgs; [
             pkg-config
           ];
         };
 
-        cargoArtifacts = craneLib.buildDepsOnly (commonArgs // {
-          pname = "minimint";
+        minimintDeps = craneLib.buildDepsOnly (commonArgs // {
+          pname = "minimint-dependencies";
         });
 
-        myCrateClippy = craneLib.cargoClippy (commonArgs // {
-          inherit cargoArtifacts;
+        minimintClippy = craneLib.cargoClippy (commonArgs // {
+          cargoArtifacts = minimintDeps;
           cargoClippyExtraArgs = "-- --deny warnings";
         });
 
-        myCrate = craneLib.buildPackage (commonArgs // {
-          inherit cargoArtifacts;
+        minimint = craneLib.buildPackage (commonArgs // {
+          cargoArtifacts = minimintDeps;
         });
 
-        myCrateCoverage = craneLib.cargoTarpaulin (commonArgs // {
-          inherit cargoArtifacts;
+        minimintCoverage = craneLib.cargoTarpaulin (commonArgs // {
+          cargoArtifacts = minimintDeps;
         });
       in
       {
-        packages.default = myCrate;
+        packages.default = minimint;
         checks = {
-         inherit
-           myCrate
-           myCrateClippy
-           myCrateCoverage;
+          inherit
+            minimint
+            minimintClippy
+            minimintCoverage;
         };
+
+        devShell =
```

Comment:
I added a dev shell, hopefully close to what `shell.nix` had. Use `nix develop` to get there.